RecoLocalTracker/SiPixelRecHits/interface/PixelCPEGenericBase.h

```diff
@@ -0,0 +1,96 @@
+#ifndef RecoLocalTracker_SiPixelRecHits_PixelCPEGenericBase_H
+#define RecoLocalTracker_SiPixelRecHits_PixelCPEGenericBase_H
+
+#include "PixelCPEBase.h"
+#include <vector>
+
+class PixelCPEGenericBase : public PixelCPEBase {
+public:
+  struct ClusterParamGeneric : ClusterParam {
+    ClusterParamGeneric(){};
+    ClusterParamGeneric(const SiPixelCluster& cl) : ClusterParam(cl) {}
+    // The truncation value pix_maximum is an angle-dependent cutoff on the
+    // individual pixel signals. It should be applied to all pixels in the
+    // cluster [signal_i = fminf(signal_i, pixmax)] before the column and row
+    // sums are made. Morris
+    int pixmx;
+
+    // These are errors predicted by PIXELAV
+    float sigmay;  // CPE Generic y-error for multi-pixel cluster
```

Comment:
btw: you may have by now realized that there is no need of all these variable as they are mutual exclusive.